build-image-src/Dockerfile-provided

```diff
@@ -58,6 +58,12 @@ ENV PATH=$PATH:/usr/local/opt/lambda-builders/bin
 # Python for it to be picked up during `sam build`
 RUN pip3 install wheel
 
+# Install Go
```

Comment:
I think we don't need to do this with provided image since current go1.x runtime already uses this, please correct me if I am wrong.

Reply:
The `go1.x` image also contains the Go RPC mode shim, so the runtime can have subtly different behavior. The most conservative migration approach to migrate from `go1.x`/AL1 to AL2 would be to do `go1.x` -> `provided` -> `provided.al2`. This way if there are issues found during testing, a user can identify runtime client behavior separately from build environment issues as a possible root cause. 

This difference for testing doesn't really matter in the context of this change, since it's a build environment, and it'd be also possible to change SAM to select the `Dockerfile-go1x` when `BuildMethod: go1.x` and `Runtime: provided` - however I'm not sure how to do this wiring if that's the way you'd rather go!